server/storage/mvcc/key_index.go

```diff
@@ -238,20 +233,15 @@ func (ki *keyIndex) keep(atRev int64, available map[Revision]struct{}) {
 		return
 	}
 
-	genIdx, revIndex := ki.doCompact(atRev, available)
-	g := &ki.generations[genIdx]
-	if !g.isEmpty() {
-		// remove any tombstone
-		if revIndex == len(g.revs)-1 && genIdx != len(ki.generations)-1 {
-			delete(available, g.revs[revIndex])
-		}
-	}
+	ki.doCompact(atRev, available)
 }
 
 func (ki *keyIndex) doCompact(atRev int64, available map[Revision]struct{}) (genIdx int, revIndex int) {
 	// walk until reaching the first revision smaller or equal to "atRev",
 	// and add the revision to the available map
 	f := func(rev Revision) bool {
+		// NOTE: Duplicate keys in one txn isn't allowed. So, there is
+		// no need to check Sub revision here.
```

Comment:
This comment is confusing. Suggest to remove it to avoid confusion.

Reply:
Updated.